core/src/androidTest/java/com/amplifyframework/core/DevMenuMainViewInstrumentationTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.core;
+
+import androidx.annotation.RawRes;
+import androidx.fragment.app.testing.FragmentScenario;
+import androidx.navigation.NavDestination;
+import androidx.navigation.Navigation;
+import androidx.navigation.testing.TestNavHostController;
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.espresso.action.ViewActions;
+import androidx.test.espresso.matcher.ViewMatchers;
+
+import com.amplifyframework.testutils.Sleep;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import static androidx.test.espresso.Espresso.onView;
+
+/**
+ * Tests the navigation behavior for {@link DevMenuMainFragment}.
+ */
+public final class DevMenuMainViewInstrumentationTest {
+    // Amount of time (in milliseconds) to wait before starting
+    // a test, to allow any animations to complete.
```

Comment:
You can also try:
```groovy
android {

  ...

  testOptions {
    animationsDisabled = true
  }
}
```
to disable animations, and prevent the need for this sleep. (Sleeps in test code are brittle, since they're outside of the deterministic flow of the source code's logic.)

It would probably be appropriate, for `amplify-android` anyway, to cook that `testOption` right into the project-level `build.gradle`. I say this because, this is not a UI-centric project, so enablement of animations isn't the ethos of what this project needs to be testing. If it were a dedicated UI library, we probably would consider more refinement with that.